library/java/io/envoyproxy/envoymobile/engine/AndroidProxyMonitor.java

```diff
@@ -35,6 +35,9 @@ private AndroidProxyMonitor(Context context, EnvoyEngine envoyEngine) {
     this.envoyEngine = envoyEngine;
     this.connectivityManager =
         (ConnectivityManager)context.getSystemService(Context.CONNECTIVITY_SERVICE);
+    // PROXY_INFO is not guaranteed to be a sticky intent so we need to trigger
+    // a manual poll of proxy settings.
+    handleProxyChange(null);
     registerReceiver(context);
```

Comment:
Since the order apparently matters here, that wasn't clear to me reading the code. Can you add a comment explaining why this order is important?

Reply:
done